app/static/scripts/components/nav.js

```diff
@@ -0,0 +1,48 @@
+class Nav{
+
+  constructor(){
+    this.navIcon = document.querySelector('.nav-icon');
+    this.mobileNavContainer = document.querySelector('.mobile-nav-container');
+    this.classNavIsOpen = 'mobile-nav-open';
+    this.classUpdateNav = 'nav-mobile-is-open';
+    this.navMobileIsOpen = false;
+    this.toggleMenu = this.toggleMenu.bind(this);
+  }
+
+  updateNavIcon(){
+    this.navMobileIsOpen?
+    this.navIcon.classList.add('open'):
+    this.navIcon.classList.remove('open');
+  }
+
+  updateNavColors(){
+    this.navMobileIsOpen?
+    document.body.classList.add(this.classUpdateNav):
+    document.body.classList.remove(this.classUpdateNav);
+  }
+
+  updateMobileMenu(){
+    this.navMobileIsOpen?
+    this.mobileNavContainer.classList.add(this.classNavIsOpen):
+    this.mobileNavContainer.classList.remove(this.classNavIsOpen);
+  }
+
```

Comment:
Again, this is cool, but you should consider condensing these functions so you aren't making so many checks against if the menu is open

```javascript
    build() {
        this.navIcon.classList.add('open);
        this.mobileNavContainer.classList.add(this.classNavIsOpen);
        document.body.classList.add(this.classUpdateNav);
    }

    remove() {
      // same thing but with the remove actions
    }

    toggleMenu() {
        this.mobileNavContainer.style.willChange = 'opacity';
        this.navMobileIsOpen = !this.navMobileIsOpen;
        this.navMobileIsOpen ? this.build() : this.remove()
        ...